earthpy/tests/test_crop_image.py

```diff
@@ -119,3 +136,99 @@ def test_crop_image_fails_empty_list(basic_image_tif, basic_geometry):
     with rio.open(basic_image_tif) as src:
         with pytest.raises(ValueError):
             es.crop_image(src, list())
+
+
+def test_crop_all_returns_list(in_paths, output_dir, basic_geometry_gdf):
+    """Test that crop all returns a list. """
+    img_list = es.crop_all(
+        in_paths, output_dir, basic_geometry_gdf, overwrite=True
+    )
+    assert type(img_list) == list
+
+
+def test_crop_all_returns_list_when_given_list(
+    in_paths, output_file_list, basic_geometry_gdf
+):
+    """Test that crop all returns a list when given a list instead of a directory. """
+    img_list = es.crop_all(
+        in_paths, output_file_list, basic_geometry_gdf, overwrite=True
+    )
+    assert isinstance(img_list, list)
+
+
+def test_crop_all_files_exist(in_paths, output_dir, basic_geometry_gdf):
+    """Test that crop all actually creates the files in the directory. """
+    img_list = es.crop_all(
+        in_paths, output_dir, basic_geometry_gdf, overwrite=True
+    )
+    for files in img_list:
+        assert os.path.exists(files)
+
+
+def test_crop_all_fails_overwrite(in_paths, output_dir, basic_geometry_gdf):
+    """Test that crop all fails when overwrite isn't set to True if files already exist. """
+    with pytest.raises(ValueError, match="The file "):
+        es.crop_all(in_paths, output_dir, basic_geometry_gdf)
+
+
+def test_crop_all_fails_bad_dir(in_paths, basic_geometry_gdf):
+    """Test crop all fails if user provides a bad directory path. """
+    bad_path = "Badpath"
+    with pytest.raises(TypeError, match="The output directo"):
+        es.crop_all(in_paths, bad_path, basic_geometry_gdf, overwrite=True)
+
+
+def test_crop_all_fails_bad_dir_list(in_paths, basic_geometry_gdf):
+    """Test crop all fails if user provides a list of bad directory path. """
+    bad_path_list = ["Badpath0", "Badpath1", "Badpath2", "Badpath3"]
+    with pytest.raises(TypeError, match="The output directo"):
+        es.crop_all(
+            in_paths, bad_path_list, basic_geometry_gdf, overwrite=True
+        )
+
+
+def test_crop_all_returns_list_of_same_len(
+    in_paths, output_dir, basic_geometry_gdf
+):
+    """Test that crop all returns a list of the same length as the input list. """
+    img_list = es.crop_all(
+        in_paths, output_dir, basic_geometry_gdf, overwrite=True
+    )
+    assert len(img_list) == len(in_paths)
+
+
+def test_crop_all_fails_with_mismatch_list_len(
+    in_paths, output_file_list, basic_geometry_gdf
+):
+    """Test that crop all fails if the input and output list lengths don't match. """
+    path = os.path.commonpath(output_file_list)
+    output_file_list.append(os.path.join(path, "file5.tif"))
+    with pytest.raises(TypeError, match="The list of input bands does not m"):
```

Comment:
rather than bands for this message, please use rasters. remember htat these could just be random images not always bands.